src/votekit/ballot_generator.py

```diff
@@ -509,84 +515,110 @@ class AlternatingCrossover(BallotGenerator):
     def __init__(
         self,
         slate_to_candidates=None,
-        bloc_crossover_rate=None,
+        cohesion_parameters=None,
         **data,
     ):
 
         # Call the parent class's __init__ method to handle common parameters
         super().__init__(**data)
 
         self.slate_to_candidates = slate_to_candidates
-        self.bloc_crossover_rate = bloc_crossover_rate
+        self.cohesion_parameters = cohesion_parameters
+
+        for bloc, pref_interval in self.pref_interval_by_bloc.items():
+            if 0 in pref_interval.values():
+                raise ValueError("In AC model, all candidates must have non-zero preference.")
 
     def generate_profile(self, number_of_ballots) -> PreferenceProfile:
 
         ballot_pool = []
 
+        # compute the number of bloc and crossover voters in each bloc using Huntington Hill
+        voter_types = [(b, type) for b in list(self.bloc_voter_prop.keys()) \
```

Comment:
`self.bloc_voter_prop.keys()` returns a list, so there's no need to cast as a list here.